### PR TITLE
Adds api's to get list of namespaces and deployments

### PR DESCRIPTION
**Pull request by bharathappali**
This PR adds the api's which helps autotune UI to populate the namespaces and deployments on selection by the user.

Signed-off-by: bharathappali <abharath@redhat.com>

### CODE REVIEW

**Comment by bharathappali** [comment]
@dinogun @msvinaykumar I have resolved the conflicts and pushed the changes. Can I please get it reviewed?

File: src/main/java/com/autotune/analyzer/Analyzer.java [hunk 1]
```diff
@@ -59,5 +59,9 @@ public static void addServlets(ServletContextHandler context) {
         context.addServlet(UpdateResults.class, ServerContext.UPDATE_RESULTS);
         context.addServlet(Recommendation.class, ServerContext.RECOMMEND_RESULTS);
         context.addServlet(PerformanceProfileService.class, ServerContext.CREATE_PERF_PROFILE);
-    }
+
+		// Adding UI support API's
+		context.addServlet(ListNamespaces.class, ServerContext.LIST_NAMESPACES);
+		context.addServlet(ListDeploymentsInNamespace.class, ServerContext.LIST_DEPLOYMENTS);
+	}
```
Comment (dinogun) [inline src/main/java/com/autotune/analyzer/Analyzer.java:66]: Seems this is misaligned

File: src/main/java/com/autotune/analyzer/services/ListDeploymentsInNamespace.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                            || inputJson.getString(AutotuneConstants.JSONKeys.NAMESPACE).isBlank()
+                            || inputJson.getString(AutotuneConstants.JSONKeys.NAMESPACE).isEmpty()) {
+                        response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                        namespace = null;
+                    } else {
+                        namespace = inputJson.getString(AutotuneConstants.JSONKeys.NAMESPACE);
+                    }
+                }
+            }
+            if (null != namespace) {
```
Comment (dinogun) [inline src/main/java/com/autotune/analyzer/services/ListDeploymentsInNamespace.java:52]: this should just be an else statement?

File: src/main/java/com/autotune/utils/ServerContext.java [hunk 3]
```diff
@@ -56,4 +56,9 @@ public class ServerContext {
     public static final String EXPERIMENT_MANAGER_LIST_EXPERIMENT_END_POINT = EXPERIMENT_MANAGER_SERVER_PORT + EXPERIMENT_MANAGER_LIST_EXPERIMENT_TRIAL;
     public static final String EXPERIMENT_MANAGER_LIST_TRIAL_STATUS = ROOT_CONTEXT + "listTrialStatus";
     public static final String EXPERIMENT_MANAGER_LIST_TRIAL_STATUS_END_POINT = EXPERIMENT_MANAGER_SERVER_PORT + EXPERIMENT_MANAGER_LIST_TRIAL_STATUS;
+
+	// UI support EndPoints
+	public static final String QUERY_CONTEXT = ROOT_CONTEXT + "query/";
+	public static final String LIST_NAMESPACES = QUERY_CONTEXT + "listNamespaces";
+	public static final String LIST_DEPLOYMENTS = QUERY_CONTEXT + "listDeployments";
```
Comment (dinogun) [inline src/main/java/com/autotune/utils/ServerContext.java:63]: Seems misaligned

File: src/main/java/com/autotune/analyzer/Analyzer.java (hunk 1 above)
Comment (bharathappali) [reply src/main/java/com/autotune/analyzer/Analyzer.java]: I tried to fix it, please lemme know if it's good now.

File: src/main/java/com/autotune/analyzer/services/ListDeploymentsInNamespace.java (hunk 2 above)
Comment (bharathappali) [reply src/main/java/com/autotune/analyzer/services/ListDeploymentsInNamespace.java]: Ah! yes you are right, Not sure why I have written in that way. I don't currently remember the reason now, but to me it looks like I have written `null != namespace` case first and for some reason I need to add a null check which I placed on top of it so it turned out to be two if cases.

Corrected it now. Lemme know if any other changes needed.

File: src/main/java/com/autotune/utils/ServerContext.java (hunk 3 above)
Comment (bharathappali) [reply src/main/java/com/autotune/utils/ServerContext.java]: I feel it's tabs vs spaces issue, It should be fixed now. I'm having tab = 4 spaces setting on my IDE

File: src/main/java/com/autotune/analyzer/services/ListDeploymentsInNamespace.java (hunk 2 above)
Comment (bharathappali) [reply src/main/java/com/autotune/analyzer/services/ListDeploymentsInNamespace.java:52]: Ah! sorry I jus overlooked earlier.

Here is why i have added `null != namespace`
- First I'll check if there is a namespace param passed
- if not then i'll check if there is a json body which has key `namespace`
- if I find namespace i'll search deployments
- if not i'll return invalid namespace

So in this way this api is flexible to accept namespace as `autotuneUrl/query/listDeployments?namespace=<namespace>` or json sent to `autotuneUrl/query/listDeployments`

**Comment by msvinaykumar** [comment]
LGTM

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**